tests/python/pants_test/pantsd/test_process_manager.py

```diff
@@ -1,15 +1,18 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
-
 import errno
 import logging
 import os
+import subprocess
+import sys
 import unittest
 import unittest.mock
 from contextlib import contextmanager
+from textwrap import dedent
 
 import psutil
 import pytest
+from _pytest.tmpdir import TempdirFactory
```

Comment:
Why use Pytest rather than `contextutil.temporary_dir()`?

Reply:
I mean - you're the guy who doesn't like nesting. I can't win! Do you really want me to use it? Just let me know the new trumping logic if so.

Reply:
> I can't win!

I'm sorry for the frustration :/

> Do you really want me to use it?

The only thing that weirded me out is the private import. Otherwise, yeah, this would be great (but either approach fine)

--

It's okay to keep as is, these both work.

Reply:
I'm going to leave as is here and merge before I dash out the door. I think the pytest tmp fixutres are more useful not because of the visual reason (flattening), but the practical reason that those tmpdirs hang around under `/tmp/pytest-of-jsirois/pytest-current/...` so tests are easier to debug.

Reply:
Cool, that is what I was trying to get at - if you had forgotten we have `.temporary_dir()` vs using Pytest for a dedicated reason. Sg